metadata_mapper/mappers/mapper.py

```diff
@@ -142,9 +142,10 @@ def enrich(self, enrichment_function_name: str, **kwargs):
         func: Callable = getattr(self, enrichment_function_name)
         try:
             return func(**kwargs)
-        except Exception as e:
-            print(f"ENRICHMENT ERROR: {str(kwargs)}")
-            raise e
+        except Exception:
+            raise Exception(f"ENRICHMENT ERROR for enrichment: `{enrichment_function_name}` "
+                f"with kwargs: `{str(kwargs)}` "
+                )
```

Comment:
This is why I want https://docs.python.org/3/library/exceptions.html#BaseException.add_note and https://docs.python.org/3/library/exceptions.html#ExceptionGroup

Reply:
I think this means we do lose the exception that was thrown by the enrichments, though? Like if it was a ValueError or a KeyError or whatnot.

Reply:
No, the exception thrown by enrichments does appear in the logs. I tested it out.